src/system.c

```diff
@@ -1673,6 +1673,59 @@ void SySetGapRootPath( const Char * string )
     return; 
 }
 
+/****************************************************************************
+**
+*F  SySetInitialGapRootPaths( <string> )  . . . . .  set the root directories
+**
+**  Setup GAP's initial root paths, based on three sources
+**  1) The environment variable GAPROOT
+**  2) If the environment variable GAPROOT is undefined, the 
+**     C macro GAPROOT_LOCATION (defined when GAP was compiled).
+**  3) Finally, try GAP's exec location.
+*/
+static void SySetInitialGapRootPaths()
+{
+    Int i,j;
+
+    if(getenv("GAPROOT")) {
+        SySetGapRootPath(getenv("GAPROOT"));
+        return;
+    }
+
+#ifdef GAPROOT_LOCATION
+/* This is just some nasty C to wrap quotes around GAPROOT_LOCATION */
+#define stringify2(s) stringify(s)
+#define stringify(s) #s
```

Comment:
But why do these tricks? Just require the developer to properly quote `GAPROOT_LOCATION` (resp. `SYS_DEFAULT_PATHS`)